### ISSUE TITLE
ID, logs, backup

### ISSUE THREAD

**Issue by MullerUA** [gensyn-ai/rl-swarm#16]
Hello,  

I'm testing the launch in Docker (CPU). I have a few questions and would like to get some answers since Discord is still closed. I appreciate your help in advance.  

1. I understand that the node identifier is the `hivemind_grpo_trainer ID` [F-fffb1e15-02......], but after restarting the container, the ID changes. How is my contribution tracked? Is it possible that it updates in each round/epoch?  

**Example:**  
2025-03-12 13:51:51,391 - main - INFO - * Starting training 2025-03-12 13:51:51 for 3.0 epochs*
INFO:__main__:*** Starting training 2025-03-12 13:51:51 for 3.0 epochs***
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:[F-fffb1e15-0228-427b-b783-ab6330f0cef5] Joining round: 14 starting at stage: 1
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:[F-fffb1e15-0228-427b-b783-ab6330f0cef5] Training round: 14 stage: 1
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:[F-fffb1e15-0228-427b-b783-ab6330f0cef5] Training round: 14 stage: 1
Generating train split: 272 examples [00:00, 10195.63 examples/s]
Map: 100%|██████████| 272/272 [00:00<00:00, 2208.16 examples/s]
  0%|          | 0/2 [00:00<?, ?it/s]`use_cache=True` is incompatible with gradient checkpointing. Setting use_cache=False.

**docker restart swarm_node**

025-03-12 14:49:42,404 - main - INFO - * Starting training 2025-03-12 14:49:42 for 3.0 epochs*
INFO:__main__:*** Starting training 2025-03-12 14:49:42 for 3.0 epochs***
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:[F-8df0345c-76b2-4a9f-9984-cd396a560b9b] Joining round: 15 starting at stage: 0
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:[F-8df0345c-76b2-4a9f-9984-cd396a560b9b] Training round: 15 stage: 0
INFO:hivemind_exp.trainer.hivemind_grpo_trainer:[F-8df0345c-76b2-4a9f-9984-cd396a560b9b] Training round: 15 stage: 0
Joining swarm with initial_peer = /ip4/38.101.215.13/tcp/30002/p2p/QmQ2gEXoPJg6iMBSUFWGzAabS2VhnzuS782Y637hGjfsRJ
  0%|          | 0/2 [00:00<?, ?it/s]`use_cache=True` is incompatible with gradient checkpointing. Setting use_cache=False.

2. What needs to be backed up in case I need to transfer the node to another server?  
3. I'm checking the logs of `swarm_node` and see that I'm in round 14, but when I open the web board, I see that round 15 is already running. This raises the question of whether the logs are not updating and why I don't see my agent participating in the new round.  

Again, thank you for your answers.

**Comment by mix33** [comment]
Precisely what I meant to ask.

**Comment by h-grieve** [comment]
Hey @MullerUA - thanks for your questions. On each:

1. We don't currently guarantee tracking of ID - but please keep your eyes peeled for future releases.
2. As above - portability of ID will be enabled in a future release.
3. In the setting where the coordinator Peer has more powerful hardware than you, it can cause your local device to lag behind. It will eventually catch-up - usually by skipping rounds. 

Hope this helps and please stay tuned!

**Comment by MullerUA** [comment]
Thanks for the reply! 
Looking forward to future releases
